packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3501,6 +3504,10 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
       if (child != null) {
         deactivateChild(child);
       }
+      assert(() {
```

Comment:
`Element.updateChild` is also evidence that `this` is being rebuilt.
Instead of doing `_debugElementWasRebuilt(context)` in `buildScope` just for `LayoutBuilder` this feels slightly better?

Reply:
> because it may potentially not have any children.

that case is already taken care of here: 
https://github.com/flutter/flutter/blob/2aa348b9407e96ffe4eca8e8f213c7984afad3f7/packages/flutter/lib/src/widgets/framework.dart#L3562-L3565

But the above line doesn't call `_debugElementWasRebuilt` on the root element `Element` that initiated the subtree update, like in `LayoutBuilder`:
https://github.com/flutter/flutter/blob/2aa348b9407e96ffe4eca8e8f213c7984afad3f7/packages/flutter/lib/src/widgets/layout_builder.dart#L135

With `_debugElementWasRebuilt`'s current implementation it's OK for it to be called on the same `Element` multiple times in a frame.

Reply:
Another reason why this feels out of place here: technically you don't have to go through `updateChild`. You can just call the methods called from here directly and bypass `updateChild` completely.

Reply:
Hm, but I guess if you do that you miss also some of the other debug key-locig at the end of this method?